paf-mvp-frontend/tests/test-cases/queue.test.ts

```diff
@@ -0,0 +1,88 @@
+import { TestWatcher } from 'jest';
+import { isExportDeclaration } from 'typescript';
+import { Command, CommandQueue, processCommands } from '../../src/utils/queue';
+
+class CommandMocker {
+  public called = false;
+
+  func(): Command {
+    return () => {
+      this.called = true;
+    };
+  }
+}
+
+describe('queue', () => {
+  let cmd1: CommandMocker;
+  let cmd2: CommandMocker;
+  let cmd3: CommandMocker;
+  const lib = {
+    queue: [] as CommandQueue,
+  };
+
+  beforeEach(() => {
+    cmd1 = new CommandMocker();
```

Comment:
you must re-initialize the queue in the `beforeEach()` otherwise the assertions are not correct.
As a matter of fact, "nested before processor" actually fails.
But I must admit I don't understand why this test exists.

Reply:
Good catch!
This actually puzzles me for solving the edge case of the nested calls.

It was to handle such nasty code. Maybe we just can't without an inversion of control which make the code less simple. I don't know if it is worth it.

````
       function fromBidWinToAuditLog(bid) {
            let pafObj = bid.meta.paf;
            if (pafObj === undefined) {
                return;
            }

            // Don't know in witch context fromBidWinToAuditLog
            // will be called. So we push in the queue to be sure
            // that it appears. 
            PAF.queue.push(() => {
                PAF.registerTransmissionResponse({
                    prebidTransactionId: bid.transactionId,
                    adUnitCode: bid.adUnitCode,
                    contentId: pafObj.contentId
                }, pafObj.transmission);
            });
        }

        var pbjs = pbjs || {};
        pbjs.que = pbjs.que || [];

        var PAF = PAF || {};
        PAF.queue = PAF.queue || [];

         pbjs.que.push(function() {
            pbjs.addAdUnits(adUnits);
            pbjs.setConfig(...);

            PAF.queue.push(() => {
                pbjs.onEvent("bidWon", fromBidWinToAuditLog);
            });

            pbjs.requestBids({
                bidsBackHandler: initAdserver,
                timeout: PREBID_TIMEOUT
            });
        });   
````

Reply:
I this case, I guess the processCommands should simply have a `while(queue not empty)` loop?

Reply:
Indeed, thanks!